packages/compass-import-export/src/import/import-csv.spec.ts

```diff
@@ -634,6 +636,45 @@ describe('importCSV', function () {
     expect(errorsText).to.equal(formatErrorLines(expectedErrors));
   });
 
+  it('errors when there are database errors (stopOnErrors=true)', async function () {
+    const lines = ['a,b', '1,2', '3,4', '5,6'];
+
+    const ns = 'db.col';
+
+    const fields = {
+      a: 'int',
+      b: 'int',
+    } as const;
+
+    const output = temp.createWriteStream();
+    const progressCallback = sinon.spy();
+    const errorCallback = sinon.spy();
+
+    await updateCollection(ns, {
+      validator: {
+        $jsonSchema: {
+          required: ['xxx'],
+        },
+      },
+    });
+
+    const promise = importCSV({
+      dataService,
+      ns,
+      fields,
+      input: Readable.from(lines.join('\n')),
```

Comment:
Somewhat unrelated random thought: Do we have a test for files with \r\n line endings somewhere?

Reply:
I've just assumed that it works because that's up to the parser lib to interpret. But I can add some more tests. Should probably add ones for BOM characters as well and weird non-utf8 charsets just to see.

Reply:
Actually I think I'm gonna merge this PR then add tests for that in a separate one because I probably need those kinds of tests for importCSV, guessFileType and analyzeCSVFields as well. So I'll just group it together.